eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/NormalStateSpec.scala

```diff
@@ -2499,7 +2499,7 @@ class NormalStateSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike with
   def testShutdown(f: FixtureParam, script_opt: Option[ByteVector]): Unit = {
     import f._
     val bobParams = bob.stateData.asInstanceOf[DATA_NORMAL].commitments.localParams
-    alice ! Shutdown(ByteVector32.Zeroes, script_opt.getOrElse(bobParams.defaultFinalScriptPubKey))
+    alice ! Shutdown(ByteVector32.Zeroes, script_opt.getOrElse(bobParams.upfrontShutdownScript_opt.getOrElse(Script.write(Script.pay2wpkh(randomKey().publicKey)))))
```

Comment:
```suggestion
    val bobData = bob.stateData.asInstanceOf[DATA_NORMAL]
    alice ! Shutdown(ByteVector32.Zeroes, bob.underlyingActor.getOrGenerateFinalScriptPubKey(bobData))
```

And remove now unused `bobParams`. The same comment applies multiple times in this file.

Reply:
using `underlyingActor` is much nicer but we should still use the provided script if there is one

Reply:
done in https://github.com/ACINQ/eclair/pull/2565/commits/a3da92a3f0bac4f0d093ce5548eb89f5b1287ed1